lib/combineNodes.js

```diff
@@ -0,0 +1,312 @@
+'use strict';
+var arrayEqual = require('array-equal');
+var Cesium = require('cesium');
+var clone = Cesium.clone;
+var defined = Cesium.defined;
+var defaultValue = Cesium.defaultValue;
+var IDENTITY = [1,0,0,0,0,1,0,0,0,0,1,0,0,0,0,1];

```

Comment:
I think it will be cleaner to use Cesium's [Matrix4](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/Matrix4.js).  Just construct it as needed from the array and then use all its helper functions.